src/common/models/url-shortener/url-shortener.ts

```diff
@@ -15,9 +15,9 @@
  */
 
 import { Instance } from "immutable-class";
-import { Binary } from "../../utils/functional/functional";
+import { Ternary } from "../../utils/functional/functional";
 
-export type UrlShortenerFn = Binary<string, any, Promise<string>>;
+export type UrlShortenerFn = Ternary<string, any, any, Promise<string>>;
```

Comment:
Probably we could replace `any` with better types. Watch out for importing server code in client module!

Reply:
Done, though I'm a TS noob, please let me know if I should study types/your conventions more.